### ISSUE TITLE
MAST download should efficiently fallback on existing downloads

### ISSUE THREAD

**Issue by pllim** [astropy/astroquery#2820]
Example use case:

```python
from astroquery.mast import Observations

files = ['jw02727-o002_t062_nircam_clear-f090w_i2d.fits',
         'jw02727-o002_t062_nircam_clear-f277w_i2d.fits'
        ]

for fn in files:
    uri = f"mast:JWST/product/{fn}"
    result = Observations.download_file(uri, local_path=fn)
```

I already have cached copies of these files.

Problem: When MAST server performance is degraded, it takes a really long time for `download_file` to complete when I just really want it to grab the cached copy. Even when successful, it would end up just grabbing the cached copy anyway.

Workaround: I have to avoid using `download_file` altogether, hunt down where that file was downloaded to, and use it directly.

Desired solution: When MAST server takes too long to respond (maybe user can set this timeout?), automatically stops and grab a cached copy. Throw a warning if you like. Throw error as before if no cached copy found.

cc @bmorris3 

Related:

* https://github.com/lightkurve/lightkurve/issues/1159

**Comment by bsipocz** [comment]
Let's not call it a "cached" copy, as it's confusing with the already existing cache, stored elsewhere from the downloaded files.

**Comment by bsipocz** [comment]
And I would say, it always fall back on the already existing downloads (unless opted out). If that's not the case, then this is clearly a bug.

**Comment by pllim** [comment]
It does, but only after a looooong wait that I know is unnecessary.

Actually I would also accept a "just get the downloaded copy, don't connect to internet at all" as solution.

**Comment by bsipocz** [comment]
> "just get the downloaded copy, don't connect to internet at all"

yes, and I think that should be the default. If not available, go for the internet. And have a keyword that would opt out.

**Comment by bsipocz** [comment]
Btw, I'm refactoring the download utilities, and upstream them to pyvo. So many parts of mast will need to be refactored to use that. (and this will be the default behaviour, I really don't want to see unnecessary repeated downloads)

**Comment by aripollak** [comment]
Maybe an interim workaround for @pllim would be to lower the MAST timeout to something like 1?  The [default](https://github.com/astropy/astroquery/blob/9387c450137dd6ed0f9c2c96df4abfcf201059a8/astroquery/mast/__init__.py#L24) seems to be 600 seconds.
